cloudferry_devlab/cloudferry_devlab/tests/testcases/test_resource_migration.py

```diff
@@ -118,18 +130,20 @@ def test_migrate_keystone_users(self):
                                                     resource_name='user',
                                                     parameter=param)
 
-    @attr(migrated_tenant=['admin', 'tenant1', 'tenant2'])
+    @attr(migrated_tenant=['admin', 'tenant1', 'tenant2', 'tenant5'])
     def test_migrate_keystone_user_tenant_roles(self):
         """Validate user's tenant roles were migrated with correct name."""
         src_users = self.filter_users()
-        src_user_names = [user.name for user in src_users]
         dst_users = self.dst_cloud.keystoneclient.users.list()
         least_user_match = False
         for dst_user in dst_users:
-            if dst_user.name not in src_user_names:
+            src_user = [user for user in src_users
+                        if user.name.lower() == dst_user.name.lower()]
+            if len(src_user) == 0:
                 continue
             least_user_match = True
-            src_user_tnt_roles = self.src_cloud.get_user_tenant_roles(dst_user)
+            src_user_tnt_roles = \
+                self.src_cloud.get_user_tenant_roles(src_user[0])
```

Comment:
I see where it comes from, and it's invalid. You should change the way you get `src_user` value to:

```python
src_user = None

for user in src_users:
    if user.name.lower() == dst_user.name.lower():
        src_user = user
        break
```

This way there will be no confusion, but you should also find a good way of figuring out what to do when `src_user` is `None`.